packages/provider-elasticsearch/src/example-types/results/highlightUtil.js

```diff
@@ -0,0 +1,71 @@
+import _ from 'lodash/fp.js'
+import F from 'futil'
+
+let highlightContentRegEx = ({ pre_tags, post_tags }) =>
+  new RegExp(`(?<=${pre_tags[0]})(.*?)(?=${post_tags[0]})`, 'g')
```

Comment:
Lookaheads/lookbehinds are generally expensive. You can avoid them by using `matchAll` instead of `match`. Ex:

```javascript
const regex = new RegExp('<em>(.*?)</em>', 'g')
const str = 'a <em>b</em> c <em>d e</em> f'
const matches = Array.from(str.matchAll(regex)).map(_.nth(1))
```

Also, avoid compiling (instantiating) regexes multiple times if it can be avoided. For example `new RegExp('<em>(.*?)</em>', 'g')` can be compiled once and used many times inside of a loop.